src/plugins/visualize/public/application/components/deprecation_vis_warning.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiCallOut, EuiLink } from '@elastic/eui';
+import { getCoreStart } from '../../services';
+
+export const LEGACY_CHARTS_LIBRARY = 'visualization:visualize:legacyChartsLibrary';
+
+export const DeprecationWarning = () => {
+  const canEditAdvancedSettings = getCoreStart().application.capabilities.advancedSettings.save;
+  const advancedSettingsLink = getCoreStart().http.basePath.prepend(
+    `/app/management/kibana/settings?query=${LEGACY_CHARTS_LIBRARY}`
+  );
```

Comment:
I think you should be using the `application` service instead of `coreStart` here, as the application service registers URL helpers like `getUrlForAppp('management', { path: `/kibana/settings?query=${LEGACY_CHARTS_LIBRARY}` })`